components/cors-mgt/org.wso2.carbon.identity.cors.mgt.core/src/main/java/org/wso2/carbon/identity/cors/mgt/core/internal/impl/CORSManagementServiceImpl.java

```diff
@@ -75,396 +55,238 @@
-            for (String originID : originIds) {
-                Resource resource = getResourceById(originID);
-                if (resource != null) {
-                    CORSOrigin corsOrigin = new ResourceToCORSOrigin().apply(resource);
-                    if (!corsOrigin.getAppIds().contains(appId)) {
-                        // CORS origin is not registered for the appId.
-                        if (log.isDebugEnabled()) {
-                            log.debug(String.format(ERROR_CODE_ORIGIN_NOT_PRESENT.getMessage(), tenantDomain,
-                                    originID));
-                        }
-                        throw handleClientException(ERROR_CODE_ORIGIN_NOT_PRESENT, tenantDomain,
-                                originID);
-                    }
-                } else {
-                    // CORS origin is not registered for the appId.
-                    if (log.isDebugEnabled()) {
-                        log.debug(String.format(ERROR_CODE_ORIGIN_NOT_PRESENT.getMessage(), tenantDomain,
-                                originID));
-                    }
-                    throw handleClientException(ERROR_CODE_ORIGIN_NOT_PRESENT, tenantDomain,
-                            originID);
-                }
-            }
-
-            // Delete CORS origins from the application.
-            for (String originID : originIds) {
-                Resource resource = getResourceById(originID);
-                // Remove the origin association for the tenant/application.
-                getConfigurationManager().deleteAttribute(CORS_ORIGIN_RESOURCE_TYPE_NAME,
-                        resource.getResourceName(), appId);
-                resource = getResourceById(originID);
-                if (resource.getAttributes().isEmpty()) {
-                    getConfigurationManager().deleteResource(CORS_ORIGIN_RESOURCE_TYPE_NAME,
-                            resource.getResourceName());
-                }
-            }
-        } catch (ConfigurationManagementException e) {
-            throw handleServerException(ERROR_CODE_CORS_DELETE, e, tenantDomain);
-        } finally {
-            FrameworkUtils.endTenantFlow();
-        }
+        // DAO layer throws an exception if CORSApplications cannot be retrieved for the corsOriginId.
+        // i.e The corsOriginId is invalid.
+        return Collections.unmodifiableList(getCORSOriginDAO().getCORSOriginApplications(corsOriginId));
     }
 
     /**
-     * Retrieve the ConfigurationManager instance from the CORSServiceHolder.
-     *
-     * @return ConfigurationManager The ConfigurationManager instance.
+     * {@inheritDoc}
      */
-    private ConfigurationManager getConfigurationManager() {
+    @Override
+    public CORSConfiguration getCORSConfiguration(String tenantDomain) throws CORSManagementServiceException {
 
-        return CORSManagementServiceHolder.getInstance().getConfigurationManager();
+        getTenantId(tenantDomain);
```

Comment:
What's the usage of this?

Reply:
Calling this function with the tenantDomain validates it.

Reply:
I added a separate function for this.